addons/binding/org.openhab.binding.homematic/src/main/java/org/openhab/binding/homematic/internal/communicator/AbstractHomematicGateway.java

```diff
@@ -846,6 +846,38 @@ public void execute() throws IOException {
         }
     }
 
+    @Override
+    public void deleteDevice(String address, boolean reset, boolean force, boolean defer) {
+        for (RpcClient<?> rpcClient : rpcClients.values()) {
+            try {
+                rpcClient.deleteDevice(getDevice(address), translateFlags(reset, force, defer));
+            } catch (HomematicClientException e) {
+                logger.debug(e.getMessage());
```

Comment:
Please provide context for all the logging statements, and this one gives a fatal error and will be solved by supplying context, also should we really discard the stacktrace?

Reply:
I think that the stacktrace is not necessary here, since this exception will only be thrown by `getDevice()` if the device is not found on the gateway. I will, however, clarify this with a code comment (and of course provide the context).

Now that I think about it, one would usually not expect this method to be called for a device that is not on the gateway. I don't consider it a failure to try to delete a device that is not on the gateway, but it might be important to see if this happens. Do you think the INFO log-level might be more appropriate here?

Reply:
I think the info makes sense 👍

Reply:
Nice, I already included this change in my latest commit. :)